internal/bminventory/inventory.go

```diff
@@ -5211,6 +5212,22 @@ func (b *bareMetalInventory) GetInfraEnvByKubeKey(key types.NamespacedName) (*co
 	return infraEnv, nil
 }
 
+func (b *bareMetalInventory) CreateHostInKubeKeyNamespace(ctx context.Context, kubeKey types.NamespacedName, host *models.Host) error {
+	log := logutil.FromContext(ctx, b.log)
+	log.Infof("Register host: %+v (KubeKey: %+v)", host, kubeKey)
+
+	hostToCreate := &common.Host{
+		Host:                    *host,
+		KubeKeyNamespace:        kubeKey.Namespace,
+		TriggerMonitorTimestamp: time.Now(),
+	}
+	log.Infof("Create host %s infra env %s", host.ID.String(), host.InfraEnvID)
```

Comment:
Should we validate the infra env actually exists before creating the host or was that done elsewhere? 
Is it valid to create a host that references an infra env that doesn't exist anymore?